### PR TITLE
chore(ci): increase consistency for Slack messages

### LINKED ISSUE #14323: Replace hardcoded workflow names with parameters
Follow up to https://github.com/magma/magma/pull/14229. In particular, see [this comment](https://github.com/magma/magma/pull/14229#pullrequestreview-1155991165).

Throughout the GitHub actions workflows, it would be good to use code something like `"${{ github.workflow }}"` to extract the name of a workflow instead of hardcoding the name. This would increase the consistency of how things are named and make updating names easier in the future. An example of where this could be done is [here](https://github.com/magma/magma/blob/master/.github/workflows/insync-checkin.yml#L60).

cc @LKreutzer

### PR DESCRIPTION
**Pull request by Neudrino**
## Summary

- Should close https://github.com/magma/magma/issues/14323

## Test Plan

- CI

### CODE REVIEW

File: .github/workflows/agw-coverage.yml [hunk 1]
```diff
@@ -120,7 +120,7 @@ jobs:
         env:
           SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
           SLACK_TITLE: "C / C++ code coverage with Bazel"
-          SLACK_USERNAME: "${{ github.workflow }}"
```
Comment (jheidbrink) [inline .github/workflows/agw-coverage.yml:123]: Why did you change this? Can we be sure that `github.workflow` will never contain characters that need quoting in Yaml?

File: .github/workflows/bazel.yml [hunk 2]
```diff
@@ -209,7 +209,7 @@ jobs:
         env:
           SLACK_WEBHOOK: ${{ secrets.SLACK_WEBHOOK }}
           SLACK_TITLE: "Bazel Build & Test Job `bazel build //...; bazel test //...` ${{ matrix.bazel-config }}"
```
Comment (jheidbrink) [inline .github/workflows/bazel.yml:211]: While issue #14323 talks only about workflow names, would it make sense to do the same for job names in the Slack title? CC @voisey

Comment (voisey) [reply .github/workflows/bazel.yml:211]: That was part of what I intended, yes. Sorry for the lack of clarity!

Comment (voisey) [reply .github/workflows/bazel.yml:211]: Basically, I think everything should be unified as far as possible

Comment (Neudrino) [reply .github/workflows/bazel.yml:211]: The thing with the job names is, that they are [not generically available inside the context](https://stackoverflow.com/questions/71593861/how-to-get-github-actions-matrix-job-name-within-job), but only the `job_id` is. :cry:

File: .github/workflows/agw-coverage.yml (hunk 1 above)
Comment (Neudrino) [reply .github/workflows/agw-coverage.yml:123]: I would boldly assume that `${{ expression }}` gives a well sanitized object back. Similar to `ls` already escaping special characters.

File: .github/workflows/bazel.yml (hunk 2 above)
Comment (Neudrino) [reply .github/workflows/bazel.yml:211]: If we want to have the job ID somewhere inside, I would do this in a separate PR, after debating to which field it should be assigned.

**Review by jheidbrink — APPROVED** [review]
